diff --git a/node_modules/sp-react-native-in-app-updates/android/src/main/java/com/sudoplz/rninappupdates/SpReactNativeInAppUpdatesModule.java b/node_modules/sp-react-native-in-app-updates/android/src/main/java/com/sudoplz/rninappupdates/SpReactNativeInAppUpdatesModule.java
index ce9b939..429ad9e 100644
--- a/node_modules/sp-react-native-in-app-updates/android/src/main/java/com/sudoplz/rninappupdates/SpReactNativeInAppUpdatesModule.java
+++ b/node_modules/sp-react-native-in-app-updates/android/src/main/java/com/sudoplz/rninappupdates/SpReactNativeInAppUpdatesModule.java
@@ -49,6 +49,7 @@ public class SpReactNativeInAppUpdatesModule extends ReactContextBaseJavaModule
         appUpdateManager = AppUpdateManagerFactory.create(reactContext);
         appUpdateManager.registerListener(this);
         reactContext.addActivityEventListener(this);
+        checkNewAppVersionState();
     }
 
     @Override
@@ -222,5 +223,34 @@ public class SpReactNativeInAppUpdatesModule extends ReactContextBaseJavaModule
         }
     }
 
+  private void startAppUpdateImmediate(AppUpdateInfo appUpdateInfo) {
+        try {
+            appUpdateManager.startUpdateFlowForResult(
+            // Pass the intent that is returned by 'getAppUpdateInfo()'.
+            appUpdateInfo,
+            // 'AppUpdateType.IMMEDIATE' Or 'AppUpdateType.FLEXIBLE'
+            AppUpdateType.IMMEDIATE,
+            // The current activity making the update request.
+            getCurrentActivity(),
+            // Include a request code to later monitor this update request.
+            IN_APP_UPDATE_REQUEST_CODE
+        );
+        } catch (IntentSender.SendIntentException e) {
+            e.printStackTrace();
+        }
+    }
 
+    private void checkNewAppVersionState() {
+        appUpdateManager
+                .getAppUpdateInfo()
+                .addOnSuccessListener(
+                        appUpdateInfo -> {
+                            //IMMEDIATE:
+                            if (appUpdateInfo.updateAvailability()
+                                    == UpdateAvailability.DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS) {
+                                // If an in-app update is already running, resume the update.
+                                startAppUpdateImmediate(appUpdateInfo);
+                            }
+                        });
+    }
 }
